data/changelog/posts/ocaml-lsp/2025-08-20-ocaml-eglot-brings-lsp-support-to-emacs.md

```diff
@@ -0,0 +1,94 @@
+tags: [ocaml-lsp, platform]
+---
+
+# Emacs Integration for OCaml LSP Server: Introducing ocaml-eglot
+
+## TL;DR
+
+`ocaml-eglot` provides full OCaml language support in Emacs through the Language Server Protocol (LSP) instead of direct Merlin integration. It offers the same features as `merlin.el` with simplified setup and enhanced capabilities like project-wide search. If you're starting fresh or want a more standardized approach, try `ocaml-eglot`, which is actively maintained. If your current `merlin.el` setup works well, you can continue using it.
+
+**Quick start**: [Install `ocaml-lsp-server`, add `ocaml-eglot` to your Emacs config](https://github.com/tarides/ocaml-eglot?tab=readme-ov-file#installation), and get the same OCaml development experience with less configuration.
+
+## What is ocaml-eglot?
+
+`ocaml-eglot` connects Emacs to `ocaml-lsp-server` using the Language Server Protocol, providing a standardized way to get OCaml language support.
+
+Since the recent versions of Emacs (29), `eglot`, an LSP client, has been shipped with Emacs. However, `merlin.el` provides more features than LSP (which is designed to be generic), so relying solely on the features of LSP and `eglot` would limit functionality. Thus, we extended the LSP server to support more features, and `ocaml-eglot` allows you to benefit from these features in Emacs.
+
+`ocaml-eglot` is a minor mode. It therefore works in conjunction with a major mode to edit Caml code. Examples of major modes include `tuareg`, `caml-mode` and the recent `neocaml`.
+
+## Who Should Use ocaml-eglot?
+
+**Use ocaml-eglot if you:**
+- Are starting fresh with OCaml in Emacs
+- Want simplified configuration with automatic setup
+- Use multiple editors and want consistent OCaml support
+- Want access to project-wide search and rename features
+- Want to rely on an actively maintained project that evolves over time
+
+**Stick with `merlin.el` if:**
+- Your current setup is working perfectly and heavily customized
+- You prefer direct Merlin communication
+
+For the moment, we don't plan to provide any special support for `merlin.el` -- unless we receive a lot of requests.
+
+## Key Benefits
+
+- **Simplified setup**: Install package, add config, start coding
+- **Same features**: All `merlin.el` functionality with identical keybindings
+- **Enhanced capabilities**: Project-wide rename, type-based search, automatic formatting
+- **Platform integration**: Works seamlessly with opam, dune, ocamlformat, and other OCaml tools
+
+## Getting Started
+
+Follow the [installation instructions in the ocaml-eglot README](https://github.com/tarides/ocaml-eglot?tab=readme-ov-file#installation).
+
+For a detailed list of features, see [the ocaml-eglot README](https://github.com/tarides/ocaml-eglot?tab=readme-ov-file#features).
+
+## Migration from merlin.el
+
+1. Install: `opam install ocaml-lsp-server ocamlformat`
+2. Replace your Merlin configuration with the `ocaml-eglot` setup above
```

Comment:
Since the configuration is not present anymore. We should change the sentence no?